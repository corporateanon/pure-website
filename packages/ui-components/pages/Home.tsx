import React, { SFC } from 'react';
import SearchInput from '../controls/SearchInput';
import { searchContext } from '../contexts/SearchContext';
import { firebaseAuthContext } from '../contexts/FirebaseAuthContext';
import { EpisodateSearchResult } from '@pure-website/types/episodateTypes';
import LoginButton from '../controls/LoginButton';
import { FirebaseAuthEffectProps } from '@pure-website/types/authTypes';
import MultiProvider from '../utils/MultiProvider';

const { Provider: SearchProvider } = searchContext;
const { Provider: FirebaseAuthProvider } = firebaseAuthContext;
import { SearchEffectProps } from '@pure-website/types/searchTypes';

interface HomeProps {
    searchResult: EpisodateSearchResult | null;
    search: SearchEffectProps;
    firebaseAuth: FirebaseAuthEffectProps;
}

const Home: SFC<HomeProps> = ({ searchResult, firebaseAuth, search }) => {
    return (
        <>
            <MultiProvider
                providers={[
                    [FirebaseAuthProvider, { value: firebaseAuth }],
                    [
                        SearchProvider,
                        {
                            value: search
                        }
                    ]
                ]}
            >
                <div>
                    <LoginButton />
                </div>
                <div>
                    <SearchInput />
                </div>
                <div>
                    {searchResult &&
                        searchResult.tv_shows.map((show, i) => {
                            return (
                                <div key={i}>
                                    <h3>{show.name}</h3>
                                    <div>Starts: {show.start_date}</div>
                                    <div>Ends: {show.end_date}</div>
                                    {show.image_thumbnail_path ? (
                                        <img src={show.image_thumbnail_path} />
                                    ) : null}
                                </div>
                            );
                        })}
                </div>
            </MultiProvider>
        </>
    );
};

export default Home;
