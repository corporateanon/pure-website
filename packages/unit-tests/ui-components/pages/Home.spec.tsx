import * as React from 'react';
import { create } from 'react-test-renderer';
import Home from '@pure-website/ui-components/pages/Home';

describe('Home page', () => {
    it('should match snapshot', () => {
        const searchResult = {
            total: 1,
            page: 1,
            pages: 1,
            tv_shows: [
                {
                    id: 3283,
                    name: 'Doctor Who',
                    permalink: 'doctor-who-1963',
                    start_date: '1963-11-23',
                    end_date: null,
                    country: 'UK',
                    network: 'BBC One',
                    status: 'Ended',
                    image_thumbnail_path:
                        'https://static.episodate.com/images/tv-show/thumbnail/3283.jpg'
                }
            ]
        };

        const tree = create(
            <Home
                searchResult={searchResult}
                actionSearch={() => {}}
                searchQuery="hello world!"
            />
        ).toJSON();

        expect(tree).toMatchSnapshot();
    });
});
