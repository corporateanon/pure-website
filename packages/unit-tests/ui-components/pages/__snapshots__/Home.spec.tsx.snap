// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home page should match snapshot 1`] = `
Array [
  <div>
    <input
      onChange={[Function]}
      type="text"
      value="hello world!"
    />
  </div>,
  <article>
    <h3>
      x
    </h3>
    <p>
      <img
        src="x"
      />
    </p>
  </article>,
  <article>
    <p>
      <strong>
        y
      </strong>
    </p>
    <p>
      date
    </p>
  </article>,
]
`;
